mitodl/open-discussions
# introduce Formik and use it for the comment creation / editing form

#### Pre-Flight checklist



#### What are the relevant tickets?

no ticket

#### What's this PR do?

introduces Formik and uses it to refactor out a `<CommentForm />` component, which lets us delete a bunch of yucky code. We were previously sharing from code from the predecessor of this component with the post editing form component - I'll swing back at some point to switch that over to Formik as well, but for now I'll leave it as-is.

#### How should this be manually tested?

Make sure you can make and edit comments, and reply to comments, without any problems. Editing a post should work as before as well.
